### ISSUE TITLE
Impossible (?) to customise DBus system config

### ISSUE THREAD

**Issue by kirelagin** [NixOS/nixpkgs#100205]
I’d like to tweak the DBus config, in my case I’d like to allow eavesdropping for debugging purposes. The standard way of doing this is to drop-in a `system-local.conf`.

However, currently it is impossible, since the entire `/etc/dbus-1/` is generated by `pkgs.makeDBusConf` as a whole. That’s a little bit unfortunate as I would really prefer to be able to just do `environment.etc."dbus-1/system-local.conf".text = "...";`.

I see two options:

1. Rework `makeDBusConf` so that, e.g. it just returns the two files that need to be put in `/etc/dbus-1/` and then make the DBus module link the files there itself.
2. Add an optional parameter to `makeDBusConf` with the text to put in `system-local.conf` and have it generate this file to, and then expose this parameter through the DBus module.
3. Same as above, but allow to modify the contents of `system.conf` instead of creating `system-local.conf`.

I’m not sure which option would be easier/cleaner. (Also it might be that I’m missing something and there is a workaround for this already.)

cc @abbradar @layus 
for reference: #22864

**Comment by worldofpeace** [comment]
We could easily just allow a `system-local.conf` and maybe have an option for it.
This is what's in the upstream config, which we actually merge with our modifications https://gitlab.freedesktop.org/dbus/dbus/-/blame/master/bus/system.conf.in#L142

This is how it looks on my system at that line
```
  <!-- This is included last so local configuration can override what's 
       in this standard file -->


```

entirely missing :grin: 

So maybe `make-system-conf.xsl` is removing it somehow.

**Comment by worldofpeace** [comment]
Yeah, it does `<xsl:copy-of select="child::node()[name() != 'include'`. We just need to add a `and name != 'ignore-missing` and it should work.

**Comment by kirelagin** [comment]
I’m sorry, maybe I didn’t make it clear. The problem I am talking about here is not some sort of DBus misconfiguration.
Currently, as far as I can tell, it is just impossible to place any additional file into `/etc/dbus-1/` due to how this directory is created.

**Comment by alxpettit** [comment]
I wrote an overlay to fix this on my machine, but it requires compiling DBus and a bunch of stuff just to embed my modifications in. I am actually tempted to try writing an activation script that does something like:

```bash
mv /etc/dbus-1 /etc/dbus-1.link
rsync -as /etc/dbus-1.link/ /etc/dbus-1/
cat "my mods" >> /etc/dbus-1/session.conf
```

Would you accept a PR for fixing this?

**Comment by benaryorg** [comment]
As far as I understand it should be possible to provide a custom package as part of `config.services.dbus.packages` which then provides e.g. a `<pkg>/share/dbus-1/system.d` directory which would then automatically be included via the XSLT magic, so you should be able to place any configuration files in there that will behave as if written verbatim in the dbus configuration in `/etc/dbus-1`, plus if you really want to overwrite the *entire* configuration you can always `environment.etc."dbus-1".source = lib.mkForce pkgs.linkFarm "my-dbus-config" {}` or something.

Am I wrong here?
I just stumbled across this and thought that either I'm missing something, or this issue is (in a way) obsolete (just with it being a little annoying to configure, but still perfectly possible and extensible without having to maintain two parallel systems).

**Comment by benaryorg** [comment]
I have just tested this:

```nix
{ pkgs, lib, ... }:
{
  services.dbus.packages = lib.mkAfter
  [
    (pkgs.linkFarm "extra-dbus-config"
    {
      "share/dbus-1/system.d/prometheus-connection-limits.conf" = pkgs.writers.writeText "prometheus-connection-limits.conf" ''
        <busconfig>
          <limit name="max_replies_per_connection">1024</limit>
        </busconfig>
      '';
    })
  ];
}
```

The problem I had was that the default limit (128) was too low for the prometheus systemd exporter and I just raised it like that and it works.
I am not sure if there are any options that *cannot* be changed like that, but I assume for most of those options you would want to override the entire directory using *lib.mkForce* as described above.

I suspect that this issue can be closed.

**Comment by PukPukov** [comment]
There definitely should just be an option for dropping system-local.conf. The discoverability of packages-mkAfter-linkFarm hack is almost zero
